Arrowgene.Ddon.GameServer/Handler/MailSystemMailGetAllItemHandler.cs

```diff
@@ -0,0 +1,95 @@
+using Arrowgene.Ddon.GameServer.Dump;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Logging;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using System.Linq;
+using System;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class MailSystemMailGetAllItemHandler : GameRequestPacketHandler<C2SMailSystemMailGetAllItemReq, S2CMailSystemGetAllItemRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(MailSystemMailGetAllItemHandler));
+
+        public MailSystemMailGetAllItemHandler(DdonGameServer server) : base(server)
+        {
+        }
+
+        public override S2CMailSystemGetAllItemRes Handle(GameClient client, C2SMailSystemMailGetAllItemReq request)
+        {
+            var pcap = new S2CMailSystemGetAllItemRes.Serializer().Read(pcap_data);
+
+            var result = new S2CMailSystemGetAllItemRes()
+            {
+                MessageId = request.MessageId,
+            };
+            
+            bool toItemBag = false;
+            switch (request.StorageType)
+            {
+                case 13: // ItemPost   StorageType = 13
+                    toItemBag = false;
+                    break;
+                case 19: // ItemBag    StorageType = 19
+                    toItemBag = true;
+                    break;
+                case 20: // StorageBox StorageType = 20
+                    toItemBag = false;
+                    break;
+            }
+
+            S2CItemUpdateCharacterItemNtc itemUpdateNtc = new S2CItemUpdateCharacterItemNtc()
+            {
+                UpdateType = 18 // value from pcap
+            };
+
+            var attachments = Server.Database.SelectAttachmentsForSystemMail(request.MessageId);
+            foreach (var attachment in attachments)
+            {
+                try
+                {
+                    if (attachment.AttachmentType == SystemMailAttachmentType.Item && !attachment.IsReceived)
+                    {
+                        itemUpdateNtc.UpdateItemList.AddRange(Server.ItemManager.AddItem(Server, client.Character, toItemBag, attachment.Param1, attachment.Param2));
+                    }
+                    attachment.IsReceived = true;
+                }
+                catch (Exception ex)
+                {
+                    // Squelch the exception.
+                    // Something happened so the attachment can't be received
+                    attachment.IsReceived = false;
```

Comment:
Wouldn't we want the response Error value to be set to some error value so the user is aware that something went wrong?

Reply:
So this is interesting. If we return this with `IsReceived` as false, it pops up a message in red text, which contains the number of items that the player was unable to claim.

Reply:
Understood, i think this is enough to make the user aware that something didn't go right